src/internal/type.ts

```diff
@@ -56,3 +62,217 @@ export interface IRequest {
 }
 
 export type ICanonicalRequest = string
+
+export interface ICredentials {
+  accessKey: string
+  secretKey: string
+  sessionToken?: string
+}
+
+export type UploadID = string
+export type LegalHoldStatus = 'ON' | 'OFF'
+export type NoResultCallback = (error: unknown | null) => void
+export type ResultCallback<T> = (error: unknown | null, result: T) => void
+export type TagList = Record<string, string>
+export type EmptyObject = Record<string, never>
+export type VersionIdentification = { versionId?: string }
+export type Lifecycle = LifecycleConfig | null | ''
+export type Lock = LockConfig | EmptyObject
+export type Retention = RetentionOptions | EmptyObject
+export type IsoDate = string
+export type GetObjectOpt = {
+  versionId?: string
```

Comment:
I am depending on small changes of this PR, like the introduction of GetObjectOpt and StatObjectOpts.

Can I open a PR in order to introduce them without waiting the end of the refactoration or have I to wait? (or is there anything else I can do?)

Reply:
The refactoration is already done, we just need to split this PR into small pieces for maintainer to review them. You can cherry-pick the part you need and create a seprated PR.